在Mac环境运行quick-start demo时，产生错误

**Describe the bug**
根据quick-start demo进行测试时，发生该问题

**Error message**

FAILED: build64_release/hello-world 
g++ -o build64_release/hello-world   -static-libgcc -static-libstdc++ @build64_release/hello-world.rsp 
clang: error: unsupported option '-static-libgcc'
ninja: build stopped: subcommand failed.
Blade(error): Build failure.
Blade(info): Cost time 0.521s
Blade(error): Failure


**System (please complete the following information):**
 - OS: MacOS 14.5 (23F79)
 - Version  blade (developing, unversioned)

please read the FAQ for the solution.

On Thu, 7 Nov 2024 at 12:48, jiahui chen ***@***.***> wrote:

> *Describe the bug*
> 根据quick-start demo进行测试时，发生该问题
>
> *Error message*
>
> FAILED: build64_release/hello-world
> g++ -o build64_release/hello-world -static-libgcc -static-libstdc++
> @build64_release/hello-world.rsp
> clang: error: unsupported option '-static-libgcc'
> ninja: build stopped: subcommand failed.
> Blade(error): Build failure.
> Blade(info): Cost time 0.521s
> Blade(error): Failure
>
> *System (please complete the following information):*
>
>    - OS: MacOS 14.5 (23F79)
>    - Version blade (developing, unversioned)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/chen3feng/blade-build/issues/1048>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAPH4SUAC5TYRVXEWIDALATZ7LWLTAVCNFSM6AAAAABRKKRTOGVHI2DSMVQWIX3LMV43ASLTON2WKOZSGYZTSOBWHAZTSNA>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>

--- a/src/blade/cc_targets.py
+++ b/src/blade/cc_targets.py
@@ -1447,7 +1447,9 @@ class CcBinary(CcTarget):
     def _generate_cc_binary_link_flags(self, dynamic_link):
         linkflags = []
         toolchain = self.blade.get_build_toolchain()
-        if not dynamic_link and toolchain.cc_is('gcc') and version_parse(toolchain.get_cc_version()) > version_parse('4.5'):
+        if (not dynamic_link and toolchain.cc_is('gcc')
+                and version_parse(toolchain.get_cc_version()) > version_parse('4.5')
+                and sys.platform != 'darwin'):
             linkflags += ['-static-libgcc', '-static-libstdc++']
         if self.attr.get('export_dynamic'):
             linkflags.append('-rdynamic')

